library/ssl_misc.h

```diff
@@ -2137,33 +2137,33 @@ static inline int mbedtls_ssl_tls13_sig_alg_for_cert_verify_is_supported(
     switch( sig_alg )
     {
 #if defined(MBEDTLS_ECDSA_C)
-#if defined(MBEDTLS_SHA256_C) && defined(MBEDTLS_ECP_DP_SECP256R1_ENABLED)
+#if defined(MBEDTLS_HAS_ALG_SHA_256_VIA_MD_OR_PSA_BASED_ON_USE_PSA) && defined(MBEDTLS_ECP_DP_SECP256R1_ENABLED)
```

Comment:
TLS 1.3 is not affected by `MBEDTLS_USE_PSA_CRYPTO` and always uses PSA, except for the parts that are common with TLS 1.2. This part (signature for certificate verification) is not common with TLS 1.2 (see the `#ifdef` guarding this function), so I think the correct macro here is `PSA_WANT_ALG_SHA_256`.